app/client/src/workers/Evaluation/JSObject/utils.ts

```diff
@@ -151,14 +150,14 @@ export const updateJSCollectionInUnEvalTree = (
         }
       } else {
         varList.push(newVar.name);
-        const reactivePaths = oldConfig.reactivePaths;
+        const reactivePaths = jsEntityConfig.reactivePaths;
         reactivePaths[newVar.name] =
           EvaluationSubstitutionType.SMART_SUBSTITUTE;
 
-        const dynamicBindingPathList = oldConfig.dynamicBindingPathList;
+        const dynamicBindingPathList = jsEntityConfig.dynamicBindingPathList;
         dynamicBindingPathList.push({ key: newVar.name });
 
-        set(modifiedUnEvalTree, `${entityName}.variables`, varList);
+        set(jsEntityConfig, `${entityName}.variables`, varList);
```

Comment:
Shouldn't we modify the `configTree` here instead of jsEntityConfig? 
```suggestion
        set(configtree, `${entityName}.variables`, varList);
```